### PR TITLE
Update kerberos login to support diacritics

### PR DESCRIPTION
**Pull request by adfoster-r7**
### Verification

Create a users.txt with the contents:

```
basic_user
wîth.dìáçriticš
```

Run the module against your DC:

```
msf6 auxiliary(scanner/kerberos/kerberos_login) > run rhost=192.168.123.13 domain=adf3.local user_file=./users.txt pass_file=./wordlist.txt verbose=true
```

#### Before

```
[*] 192.168.123.13 - User: "w\xC3\xAEth.d\xC3\xAC\xC3\xA1\xC3\xA7ritic\xC5\xA1" user not found
```

#### After

```
[*] 192.168.123.13 - User: 'wîth.dìáçriticš' user not found
```

### CODE REVIEW

**Comment by sempervictus** [comment]
@adfoster-r7 - for educational purposes, happen to know _why_ does `format_user(u)` works when `u.inspect` doesnt?
This smells like some Rust `impl fmt::Display for User` madness at first glance :smile:

**Comment by sempervictus** [comment]
Ah... thanks. There be dragons down there - for what seems like years, we were fighting the various encoding artifacts resulting from MRI's evolving interpretation of the subject. 1.8 was OK, 1.9 was a bloodbath, and the forceful encoding hacks finally settled things down in what i recall to be the MRI2.x timeframe.

That said, i wonder how prevalent this problem actually is across the board - non-Latin charsets shouldn't prevent exploitation or enumeration, and if we're barfing on encoded outputs then it might make sense to have something (maybe a bit more explicit in its written form) generic for formatting outputs to DB or console (since DB encoding is also a thing).

**Comment by adfoster-r7** [comment]
It's before my time; But I believe the ascii encoding was added to all strings to make writing shellcode easier

For instance ruby raises exceptions if you try to concatenate utf-8 and ascii strings

```ruby
datastore = { 'USER_INPUT' => "wîth.dìáçriticš".force_encoding('utf-8') }
shellcode = "\xcc".force_encoding('ASCII-8BIT') + datastore['USER_INPUT']

=> (irb):20:in `+': incompatible character encodings: ASCII-8BIT and UTF-8 (Encoding::CompatibilityError)
```

File: docs/navigation.rb [hunk 1]
```diff
@@ -63,6 +63,10 @@ def without_prefix(prefix)
       {
         path: 'Metasploit-Guide-Kubernetes.md',
         title: without_prefix('Metasploit Guide ')
+      },
+      {
+        path: 'kerberoasting.md',
+        title: 'Kerberoasting'
```
Comment (adfoster-r7) [inline docs/navigation.rb:69]: Change unrelated to this PR, just needed to fix the docs failing

**Comment by cgranleese-r7** [comment]
Looks good to me 👍 
image

**Pull request merged**